### PR TITLE
TimeDistributed : unroll RNN when using TF backend

### PR DESCRIPTION
**Pull request by farizrahman4u**
TF dynamic rnn not working with ndim > 3

#3830

### CODE REVIEW

**Comment by carlthome** [comment]
Hmm, this seems related https://github.com/fchollet/keras/issues/3596

File: keras/layers/wrappers.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def step(x, states):
                 return output, []
 
             last_output, outputs, states = K.rnn(step, X,
-                                                 initial_states=[])
+                                                 initial_states=[], unroll=K._BACKEND == 'tensorflow')
```
Comment (fchollet) [inline keras/layers/wrappers.py]: The correct way to access the current backend is `K.backend() == `

Better to set an `unroll` variable outside.

With `unroll=True` we need to make sure that the number of timesteps is defined, so we will need a check and exception in that case.

Comment (farizrahman4u) [inline keras/layers/wrappers.py]: What should be the error message when input length is None?

File: keras/layers/wrappers.py [hunk 2]
```diff
@@ -112,9 +112,15 @@ def call(self, X, mask=None):
             def step(x, states):
                 output = self.layer.call(x)
                 return output, []
-
+            input_length = input_shape[1]
+            if K.backend() == 'tensorflow' and len(input_shape) > 3:
+                if input_length is None:
+                    raise Exception('Number of timesteps should be defined for unrolling the RNN')
```
Comment (fchollet) [inline keras/layers/wrappers.py]: See: https://github.com/fchollet/keras/blob/b184c76205c6811fe8e8aad59abba6c4aff89c42/keras/layers/wrappers.py#L91

**Comment by fchollet** [comment]
LGTM, thanks

**Pull request merged**